File: output/page_file_generator/old_files/Equfix_Place_On_Alert_selenium.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import java.time.Duration;
import java.util.Set;

public class EquifaxPage {

    private WebDriver driver;
    private WebDriverWait wait;

    @FindBy(how = How.XPATH, using = "//button[@id='continue-button']")
    private WebElement continueButton;

    @FindBy(how = How.XPATH, using = "//input[@id='zip']")
    private WebElement zip;

    @FindBy(how = How.XPATH, using = "//button[@id='efx-dropdown-label-753393']")
    private WebElement dropdownCity;

    @FindBy(how = How.XPATH, using = "//input[@id='city']")
    private WebElement city;

    @FindBy(how = How.XPATH, using = "//input[@id='addressLine2Id']")
    private WebElement addressLine2;

    @FindBy(how = How.XPATH, using = "//input[@id='address']")
    private WebElement addressLine1;

    @FindBy(how = How.XPATH, using = "//input[@id='phoneNumber']")
    private WebElement phoneNumber;

    @FindBy(how = How.XPATH, using = "//input[@id='ssn']")
    private WebElement ssn;

    @FindBy(how = How.XPATH, using = "//input[@id='dob']")
    private WebElement dob;

    @FindBy(how = How.XPATH, using = "//input[@id='lastName']")
    private WebElement lastName;

    @FindBy(how = How.XPATH, using = "//input[@id='firstNameId']")
    private WebElement firstName;

    public EquifaxPage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    public void waitForElement(WebElement element) {
        wait.until(ExpectedConditions.visibilityOf(element));
    }

    public void click(WebElement element) {
        waitForElement(element);
        element.click();
    }

    public void enterText(WebElement element, String text) {
        waitForElement(element);
        element.clear();
        element.sendKeys(text);
    }

    public void switchToWindowByIndex(int index) {
        Set<String> windowHandles = driver.getWindowHandles();
        String[] handles = windowHandles.toArray(new String[0]);
        if (index < windowHandles.size()) {
            driver.switchTo().window(handles[index]);
        } else {
            throw new RuntimeException("Invalid window index: " + index);
        }
    }

    public void switchToWindowByHandle(String handle) {
        driver.switchTo().window(handle);
    }

    public void switchToWindowMatchingUrl(String url) {
        for (String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
            if (driver.getCurrentUrl().equals(url)) {
                return;
            }
        }
        throw new RuntimeException("No window matching URL: " + url);
    }

    public void switchToNewWindow(String expectedUrl) {
        String currentHandle = driver.getWindowHandle();
        Set<String> windowHandles = driver.getWindowHandles();
        for (String handle : windowHandles) {
            if (!handle.equals(currentHandle)) {
                driver.switchTo().window(handle);
                if (driver.getCurrentUrl().contains(expectedUrl)) {
                    return;
                }
            }
        }
        throw new RuntimeException("Failed to switch to new window with URL: " + expectedUrl);
    }

    public void clickContinueButton() {
        click(continueButton);
    }

    public void enterZip(String zipCode) {
        enterText(zip, zipCode);
    }

    public void clickDropdownCity() {
        click(dropdownCity);
    }

    public void enterCity(String cityName) {
        enterText(city, cityName);
    }

    public void enterAddressLine2(String address) {
        enterText(addressLine2, address);
    }

    public void enterAddressLine1(String address) {
        enterText(addressLine1, address);
    }

    public void enterPhoneNumber(String phone) {
        enterText(phoneNumber, phone);
    }

    public void clickSsn() {
        click(ssn);
    }

    public void enterSsn(String ssnValue) {
        enterText(ssn, ssnValue);
    }

    public void clickLastName() {
        click(lastName);
    }

    public void enterLastName(String lastNameValue) {
        enterText(lastName, lastNameValue);
    }

    public void clickFirstName() {
        click(firstName);
    }

    public void enterFirstName(String firstNameValue) {
        enterText(firstName, firstNameValue);
    }
}

File: output/page_file_generator/old_files/Equfix_Place_On_Alert_selenium.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class PageObjectModel:

    LOCATORS = {
        "continue_button": (By.XPATH, "//button[@id='continue-button']"),
        "zip": (By.XPATH, "//input[@id='zip']"),
        "efx_dropdown_label": (By.XPATH, "//button[@id='efx-dropdown-label-753393']"),
        "city": (By.XPATH, "//input[@id='city']"),
        "address_line2": (By.XPATH, "//input[@id='addressLine2Id']"),
        "address": (By.XPATH, "//input[@id='address']"),
        "phone_number": (By.XPATH, "//input[@id='phoneNumber']"),
        "ssn": (By.XPATH, "//input[@id='ssn']"),
        "dob": (By.XPATH, "//input[@id='dob']"),
        "last_name": (By.XPATH, "//input[@id='lastName']"),
        "first_name_id": (By.XPATH, "//input[@id='firstNameId']")
    }

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)
        
    def wait_for_element(self, locator):
        return self.wait.until(EC.presence_of_element_located(locator))

    def click(self, locator):
        element = self.wait_for_element(locator)
        element.click()
        
    def enter_text(self, locator, text):
        element = self.wait_for_element(locator)
        element.clear()
        element.send_keys(text)
        
    def switch_to_window_by_index(self, index):
        handles = self.driver.window_handles
        self.driver.switch_to.window(handles[index])

    def switch_to_window_by_handle(self, handle):
        self.driver.switch_to.window(handle)

    def switch_to_window_matching_url(self, url):
        handles = self.driver.window_handles
        for handle in handles:
            self.driver.switch_to.window(handle)
            if self.driver.current_url == url:
                return
        raise RuntimeError(f"No window found with URL: {url}")

    def switch_to_new_window(self, url=None):
        handles = self.driver.window_handles
        self.driver.switch_to.window(handles[-1])
        if url and self.driver.current_url != url:
            raise RuntimeError(f"Switched to new window, but URL is not as expected. Expected: {url}, but found: {self.driver.current_url}")

    def click_continue_button(self):
        self.click(self.LOCATORS["continue_button"])

    def enter_zip(self, text):
        self.enter_text(self.LOCATORS["zip"], text)

    def click_efx_dropdown(self):
        self.click(self.LOCATORS["efx_dropdown_label"])

    def enter_city(self, text):
        self.enter_text(self.LOCATORS["city"], text)

    def enter_address_line2(self, text):
        self.enter_text(self.LOCATORS["address_line2"], text)

    def enter_address(self, text):
        self.enter_text(self.LOCATORS["address"], text)

    def enter_phone_number(self, text):
        self.enter_text(self.LOCATORS["phone_number"], text)

    def click_ssn(self):
        self.click(self.LOCATORS["ssn"])

    def enter_ssn(self, text):
        self.enter_text(self.LOCATORS["ssn"], text)

    def click_last_name(self):
        self.click(self.LOCATORS["last_name"])

    def enter_last_name(self, text):
        self.enter_text(self.LOCATORS["last_name"], text)

    def click_phone_number(self):
        self.click(self.LOCATORS["phone_number"])

    def enter_dob(self, text):
        self.enter_text(self.LOCATORS["dob"], text)

    def click_address_line1(self):
        self.click(self.LOCATORS["address"])

    def enter_address_line1(self, text):
        self.enter_text(self.LOCATORS["address"], text)

    def click_city_name(self):
        self.click(self.LOCATORS["city"])

    def enter_city_name(self, text):
        self.enter_text(self.LOCATORS["city"], text)

    def click_address_line2(self):
        self.click(self.LOCATORS["address_line2"])

    def enter_address_line2(self, text):
        self.enter_text(self.LOCATORS["address_line2"], text)

    def click_zip_code(self):
        self.click(self.LOCATORS["zip"])

    def enter_zip_code(self, text):
        self.enter_text(self.LOCATORS["zip"], text)